source/common/http/utility.cc

```diff
@@ -623,5 +626,48 @@ std::string Utility::PercentEncoding::decode(absl::string_view encoded) {
   return decoded;
 }
 
+const std::tuple<bool, std::string, uint32_t> Utility::ParseAuthority(const std::string& host,
```

Comment:
prefer struct than tuple (esp. >= 3 elements), also it should return string_view (perf)

```suggestion
struct ParseAuthorityResult {...};

ParseAuthorityResult Utility::parseAuthority(absl::string_view host,
```

Reply:
I can't apply absl::string_view because of this confusing behavior, detail is in https://github.com/abseil/abseil-cpp/issues/403

chunked string reproduced with `string_view::substr()` can't consider string length when we use `string_view::data()` that is called from copied string with substr function. Because of this, It will not behave correctly with `inet_pton()`, that needs `const char*` retrieved from string_view

Reply:
That's not absl::string_view issue, you should always use data() with size(). though `inet_pton()` expecting a NUL-terminated char*, so you need make a copy before there.